### ISSUE TITLE
Proxy support

### ISSUE THREAD

**Issue by hoshins** [adirh3/Fluent-Search#76]
**Is your feature request related to a problem? Please describe.**
Need to support proxy

**Describe the solution you'd like**
- Provides setting for proxy

**Describe alternatives you've considered**
- Use system configured proxy

**Additional context**
N/A

**Comment by Metroid-Prime** [comment]
I see proxy support in a lot of apps, and this is not a ding against the request, but @hoshins  how do you see proxy support benefiting Fluent Search?

The only network access I can think of is:
1. Network indexing of files
2. Checking & downloading for updates

**Comment by hoshins** [comment]
> I see proxy support in a lot of apps, and this is not a ding against the request, but @hoshins how do you see proxy support benefiting Fluent Search?
> 
> The only network access I can think of is:
> 
> 1. Network indexing of files
> 2. Checking & downloading for updates

You're right. it's for people who is behind proxy to check & download updates or plugins

**Comment by Metroid-Prime** [comment]
I don't need this, but I do support it. Keeping FS updated a core function of the app and should be expanded to users behind proxies (which I assume is many since I see proxy support in other apps).

**Comment by adirh3** [comment]
@hoshins Can you let me know how did you configure your proxy? Through Windows settings? FS should be using the default IE proxy

**Comment by hoshins** [comment]
> @hoshins Can you let me know how did you configure your proxy? Through Windows settings? FS should be using the default IE proxy

I just use proxy configuration by default IE proxy.

**Comment by adirh3** [comment]
> > @hoshins Can you let me know how did you configure your proxy? Through Windows settings? FS should be using the default IE proxy
> 
> I just use proxy configuration by default IE proxy.

Which parts of Fluent Search does not work for you? Updating version and plugins? Does the Translator plugin work for you?

**Comment by hoshins** [comment]
> > > @hoshins Can you let me know how did you configure your proxy? Through Windows settings? FS should be using the default IE proxy
> > 
> > 
> > I just use proxy configuration by default IE proxy.
> 
> Which parts of Fluent Search does not work for you? Updating version and plugins? Does the Translator plugin work for you?

Both are not work for me

**Comment by hoshins** [comment]
any propgress on this?

**Comment by adirh3** [comment]
> any propgress on this?

I guess that means it still doesn't work, I did try adding support for using Windows current proxy which worked on my machine, forgot to update here.
Would configuring proxy manually in settings make sense? TBH I am not sure how it's done in other apps

**Comment by hoshins** [comment]
> > any propgress on this?
> 
> 
> 
> I guess that means it still doesn't work, I did try adding support for using Windows current proxy which worked on my machine, forgot to update here.
> 
> Would configuring proxy manually in settings make sense? TBH I am not sure how it's done in other apps

aAs far as i know typocally application provides options
- manually set host/port (id/pass optionally)
- using windows default proxy used in ie/edge

**Comment by adirh3** [comment]
> > > any propgress on this?
> > 
> > 
> > I guess that means it still doesn't work, I did try adding support for using Windows current proxy which worked on my machine, forgot to update here.
> > Would configuring proxy manually in settings make sense? TBH I am not sure how it's done in other apps
> 
> aAs far as i know typocally application provides options
> 
> * manually set host/port (id/pass optionally)
> * using windows default proxy used in ie/edge

In your case, the default one usually works for you? Or you need to configure manually?

**Comment by hoshins** [comment]
> > > > any propgress on this?
> 
> > > 
> 
> > > 
> 
> > > I guess that means it still doesn't work, I did try adding support for using Windows current proxy which worked on my machine, forgot to update here.
> 
> > > Would configuring proxy manually in settings make sense? TBH I am not sure how it's done in other apps
> 
> > 
> 
> > aAs far as i know typocally application provides options
> 
> > 
> 
> > * manually set host/port (id/pass optionally)
> 
> > * using windows default proxy used in ie/edge
> 
> 
> 
> In your case, the default one usually works for you? Or you need to configure manually?

i prefer using windows default proxy. (i just set host/port in ie and it works for apps i'm using)
